pam.c

```diff
@@ -12,6 +12,11 @@
 static char *pw_buf = NULL;
 
 void initialize_pw_backend(void) {
+	if (getuid() != geteuid() || getgid() != getegid()) {
+		swaylock_log(LOG_ERROR,
+			"swaylock has suid but doesn't require it with PAM backend");
```

Comment:
I think this would be clearer: "swaylock is setuid, but was compiled with the PAM backend. chmod a-x $argv[0] to fix. Aborting.".

Reply:
Should I pass `argv` into `initialize_pw_backend` or is there another way?

Reply:
Yeah, I would.

Reply:
Wait, did you mean `a-s`?

Reply:
Yeah, derp